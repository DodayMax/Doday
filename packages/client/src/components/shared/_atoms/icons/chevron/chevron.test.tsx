import * as React from 'react';
import { shallow } from 'enzyme';
import { Icons } from '@components';

describe('Chevron', () => {
  it('renders correctly without props', () => {
    const chevron = shallow(<Icons.Chevron />);

    expect(chevron).toMatchSnapshot();
  });

  it('renders correctly with right prop', () => {
    const chevron = shallow(<Icons.Chevron right />);

    expect(chevron).toMatchSnapshot();
  });
});