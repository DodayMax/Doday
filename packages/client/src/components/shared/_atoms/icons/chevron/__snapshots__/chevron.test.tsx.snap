// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chevron renders correctly with right prop 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    right={true}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <defs>
          <path
            d="M9.002 11.995l6.518-6.532a1.424 1.424 0 0 0-.616-2.419 1.417 1.417 0 0 0-1.39.409L5 11.986l2.185 2.19 6.358 6.371a1.417 1.417 0 0 0 2.413-.617 1.424 1.424 0 0 0-.408-1.393l-6.546-6.542z"
            id="chevron-1"
          />
        </defs>,
        <g
          fill="none"
          fillRule="evenodd"
        >
          <path
            d="M0 0h24v24H0z"
            fill="none"
          />
          <use
            fill="#000"
            xlinkHref="#chevron-1"
          />
        </g>,
      ],
      "className": "flip",
      "height": "20",
      "viewBox": "0 0 24 24",
      "width": "20",
      "xmlns": "http://www.w3.org/2000/svg",
      "xmlnsXlink": "http://www.w3.org/1999/xlink",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <path
            d="M9.002 11.995l6.518-6.532a1.424 1.424 0 0 0-.616-2.419 1.417 1.417 0 0 0-1.39.409L5 11.986l2.185 2.19 6.358 6.371a1.417 1.417 0 0 0 2.413-.617 1.424 1.424 0 0 0-.408-1.393l-6.546-6.542z"
            id="chevron-1"
          />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "d": "M9.002 11.995l6.518-6.532a1.424 1.424 0 0 0-.616-2.419 1.417 1.417 0 0 0-1.39.409L5 11.986l2.185 2.19 6.358 6.371a1.417 1.417 0 0 0 2.413-.617 1.424 1.424 0 0 0-.408-1.393l-6.546-6.542z",
            "id": "chevron-1",
          },
          "ref": null,
          "rendered": null,
          "type": "path",
        },
        "type": "defs",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />,
            <use
              fill="#000"
              xlinkHref="#chevron-1"
            />,
          ],
          "fill": "none",
          "fillRule": "evenodd",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "d": "M0 0h24v24H0z",
              "fill": "none",
            },
            "ref": null,
            "rendered": null,
            "type": "path",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "fill": "#000",
              "xlinkHref": "#chevron-1",
            },
            "ref": null,
            "rendered": null,
            "type": "use",
          },
        ],
        "type": "g",
      },
    ],
    "type": "svg",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <defs>
            <path
              d="M9.002 11.995l6.518-6.532a1.424 1.424 0 0 0-.616-2.419 1.417 1.417 0 0 0-1.39.409L5 11.986l2.185 2.19 6.358 6.371a1.417 1.417 0 0 0 2.413-.617 1.424 1.424 0 0 0-.408-1.393l-6.546-6.542z"
              id="chevron-1"
            />
          </defs>,
          <g
            fill="none"
            fillRule="evenodd"
          >
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
            <use
              fill="#000"
              xlinkHref="#chevron-1"
            />
          </g>,
        ],
        "className": "flip",
        "height": "20",
        "viewBox": "0 0 24 24",
        "width": "20",
        "xmlns": "http://www.w3.org/2000/svg",
        "xmlnsXlink": "http://www.w3.org/1999/xlink",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <path
              d="M9.002 11.995l6.518-6.532a1.424 1.424 0 0 0-.616-2.419 1.417 1.417 0 0 0-1.39.409L5 11.986l2.185 2.19 6.358 6.371a1.417 1.417 0 0 0 2.413-.617 1.424 1.424 0 0 0-.408-1.393l-6.546-6.542z"
              id="chevron-1"
            />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "d": "M9.002 11.995l6.518-6.532a1.424 1.424 0 0 0-.616-2.419 1.417 1.417 0 0 0-1.39.409L5 11.986l2.185 2.19 6.358 6.371a1.417 1.417 0 0 0 2.413-.617 1.424 1.424 0 0 0-.408-1.393l-6.546-6.542z",
              "id": "chevron-1",
            },
            "ref": null,
            "rendered": null,
            "type": "path",
          },
          "type": "defs",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />,
              <use
                fill="#000"
                xlinkHref="#chevron-1"
              />,
            ],
            "fill": "none",
            "fillRule": "evenodd",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "d": "M0 0h24v24H0z",
                "fill": "none",
              },
              "ref": null,
              "rendered": null,
              "type": "path",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "fill": "#000",
                "xlinkHref": "#chevron-1",
              },
              "ref": null,
              "rendered": null,
              "type": "use",
            },
          ],
          "type": "g",
        },
      ],
      "type": "svg",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;

exports[`Chevron renders correctly without props 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <defs>
          <path
            d="M9.002 11.995l6.518-6.532a1.424 1.424 0 0 0-.616-2.419 1.417 1.417 0 0 0-1.39.409L5 11.986l2.185 2.19 6.358 6.371a1.417 1.417 0 0 0 2.413-.617 1.424 1.424 0 0 0-.408-1.393l-6.546-6.542z"
            id="chevron-1"
          />
        </defs>,
        <g
          fill="none"
          fillRule="evenodd"
        >
          <path
            d="M0 0h24v24H0z"
            fill="none"
          />
          <use
            fill="#000"
            xlinkHref="#chevron-1"
          />
        </g>,
      ],
      "className": "",
      "height": "20",
      "viewBox": "0 0 24 24",
      "width": "20",
      "xmlns": "http://www.w3.org/2000/svg",
      "xmlnsXlink": "http://www.w3.org/1999/xlink",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <path
            d="M9.002 11.995l6.518-6.532a1.424 1.424 0 0 0-.616-2.419 1.417 1.417 0 0 0-1.39.409L5 11.986l2.185 2.19 6.358 6.371a1.417 1.417 0 0 0 2.413-.617 1.424 1.424 0 0 0-.408-1.393l-6.546-6.542z"
            id="chevron-1"
          />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "d": "M9.002 11.995l6.518-6.532a1.424 1.424 0 0 0-.616-2.419 1.417 1.417 0 0 0-1.39.409L5 11.986l2.185 2.19 6.358 6.371a1.417 1.417 0 0 0 2.413-.617 1.424 1.424 0 0 0-.408-1.393l-6.546-6.542z",
            "id": "chevron-1",
          },
          "ref": null,
          "rendered": null,
          "type": "path",
        },
        "type": "defs",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />,
            <use
              fill="#000"
              xlinkHref="#chevron-1"
            />,
          ],
          "fill": "none",
          "fillRule": "evenodd",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "d": "M0 0h24v24H0z",
              "fill": "none",
            },
            "ref": null,
            "rendered": null,
            "type": "path",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "fill": "#000",
              "xlinkHref": "#chevron-1",
            },
            "ref": null,
            "rendered": null,
            "type": "use",
          },
        ],
        "type": "g",
      },
    ],
    "type": "svg",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <defs>
            <path
              d="M9.002 11.995l6.518-6.532a1.424 1.424 0 0 0-.616-2.419 1.417 1.417 0 0 0-1.39.409L5 11.986l2.185 2.19 6.358 6.371a1.417 1.417 0 0 0 2.413-.617 1.424 1.424 0 0 0-.408-1.393l-6.546-6.542z"
              id="chevron-1"
            />
          </defs>,
          <g
            fill="none"
            fillRule="evenodd"
          >
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
            <use
              fill="#000"
              xlinkHref="#chevron-1"
            />
          </g>,
        ],
        "className": "",
        "height": "20",
        "viewBox": "0 0 24 24",
        "width": "20",
        "xmlns": "http://www.w3.org/2000/svg",
        "xmlnsXlink": "http://www.w3.org/1999/xlink",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <path
              d="M9.002 11.995l6.518-6.532a1.424 1.424 0 0 0-.616-2.419 1.417 1.417 0 0 0-1.39.409L5 11.986l2.185 2.19 6.358 6.371a1.417 1.417 0 0 0 2.413-.617 1.424 1.424 0 0 0-.408-1.393l-6.546-6.542z"
              id="chevron-1"
            />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "d": "M9.002 11.995l6.518-6.532a1.424 1.424 0 0 0-.616-2.419 1.417 1.417 0 0 0-1.39.409L5 11.986l2.185 2.19 6.358 6.371a1.417 1.417 0 0 0 2.413-.617 1.424 1.424 0 0 0-.408-1.393l-6.546-6.542z",
              "id": "chevron-1",
            },
            "ref": null,
            "rendered": null,
            "type": "path",
          },
          "type": "defs",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />,
              <use
                fill="#000"
                xlinkHref="#chevron-1"
              />,
            ],
            "fill": "none",
            "fillRule": "evenodd",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "d": "M0 0h24v24H0z",
                "fill": "none",
              },
              "ref": null,
              "rendered": null,
              "type": "path",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "fill": "#000",
                "xlinkHref": "#chevron-1",
              },
              "ref": null,
              "rendered": null,
              "type": "use",
            },
          ],
          "type": "g",
        },
      ],
      "type": "svg",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
